### ISSUE TITLE
Sorting in Markers list seems off

### ISSUE THREAD

**Issue by MvBonin** [coderedart/jokolay#47]
Hi. 
I imported tekkit Markers and was impressed on how nice and easy it works.

When looking at the list of Markers however, the Sorting/Order seems to be a bit mixed up.

![image](https://github.com/coderedart/jokolay/assets/41330432/5ca7dbcb-8a8c-4313-bf7d-66e86c62d984)

**Comment by coderedart** [comment]
yeah. seems like it. I did remember it all being sorted when i was first working on it, i guess i messed it up somewhere while working on it. 

I will fix this in 0.3 version

**Comment by raphaelroosz** [comment]
[various_fix_202402.txt](https://github.com/coderedart/jokolay/files/14388581/various_fix_202402.txt)

Hello,

Thank you for the good work.

You'll find enclosed a patch that does a few fix:

1. order fix: BTree, HashMap, HashSet, sort or shuffle the categories in unexpected order: replaced by the OrderedHash{Map,Set} + a sort on xmls. That data structure preserve insertion order when iterating on it.
2. log flood of a few Gb: above out of order issue may lead to key not found and a logging of key/value at each failed lookup. Now when there is a failure, a boolean is updated and once sequence is finished display existing keys (and not values) for comparison.
3. allow boolean as text in addition to number so that "true" and "1" are both valid and mean the same (and "false" and "0").
4. change file extension check to actual file extension, and not just the end of the file name (so that .xml and .notxml are not confused)
5. categories without DisplayName now take the name as default substitute: this avoid to have empty rows in some packs, especially in numbers.

With the above fix, I can only find 3 remaining problems:
1. Route tags are not implemented (check Taco default pack for "floor is lava").
2. when a texture lookup fails, knowing which file it was required from would be nice (working on that now).
3. the few Gb logs issue make me realize the loading/save of package is sync and make game unresponsive since it is "hidden" under the overlay which is busy logging. Other windows are responsive.

**Comment by coderedart** [comment]
Unfortunately I can't really accept the patch now. I am moving to a custom gui framework, with wasm (and lua) plugins scripting using wasmtime. So, there's been a **lot** of churn in my local repo. 

But fortunately, this should also solve the issue with loading/saving of package. Plugins will run in non-main threads, and won't affect the main thread which is reserved for UI + rendering + event loop. And Markers functionality *will* be moved to plugins. 

I will try my best to do an alpha release with the new changes on sunday, and then tackle the problem with marker manager. The advantage is that I can work on markers and send updates as a wasm plugin over the internet, without having to update jokolay. That should also make it faster to iterate.

**Comment by MvBonin** [comment]
@raphaelroosz Just for my interest: Why didn't you do a pull request?
